crates/dojo-lang/src/manifest.rs

```diff
@@ -19,54 +36,101 @@ pub enum ManifestError {
 }
 
 /// Component member.
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Default, Debug, Serialize, Deserialize)]
 pub struct Member {
     pub name: String,
     #[serde(rename = "type")]
     pub ty: String,
 }
 
 /// Represents a declaration of a component.
-#[derive(Debug, Serialize, Deserialize)]
+#[serde_as]
+#[derive(Default, Debug, Serialize, Deserialize)]
 pub struct Component {
     pub name: String,
     pub members: Vec<Member>,
+    #[serde_as(as = "UfeHex")]
+    pub class_hash: FieldElement,
 }
 
 /// System input ABI.
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Default, Debug, Serialize, Deserialize)]
 pub struct Input {
     pub name: String,
     #[serde(rename = "type")]
     pub ty: String,
 }
 
 /// System Output ABI.
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Default, Debug, Serialize, Deserialize)]
 pub struct Output {
     #[serde(rename = "type")]
     pub ty: String,
 }
 
 /// Represents a declaration of a system.
-#[derive(Debug, Serialize, Deserialize)]
+#[serde_as]
+#[derive(Default, Debug, Serialize, Deserialize)]
 pub struct System {
     pub name: SmolStr,
     pub inputs: Vec<Input>,
     pub outputs: Vec<Output>,
+    #[serde_as(as = "UfeHex")]
+    pub class_hash: FieldElement,
     pub dependencies: Vec<String>,
 }
 
-/// Represents a declaration of a system.
-#[derive(Debug, Serialize, Deserialize)]
+#[serde_as]
+#[derive(Default, Debug, Serialize, Deserialize)]
+pub struct Contract {
+    pub name: SmolStr,
+    #[serde_as(as = "UfeHex")]
+    pub class_hash: FieldElement,
+}
+
+#[serde_as]
+#[derive(Default, Debug, Serialize, Deserialize)]
 pub struct Manifest {
+    #[serde_as(as = "UfeHexOption")]
+    pub world: Option<FieldElement>,
+    #[serde_as(as = "UfeHexOption")]
+    pub store: Option<FieldElement>,
+    #[serde_as(as = "UfeHexOption")]
+    pub indexer: Option<FieldElement>,
+    #[serde_as(as = "UfeHexOption")]
+    pub executor: Option<FieldElement>,
     pub components: Vec<Component>,
     pub systems: Vec<System>,
+    pub contracts: Vec<Contract>,
 }
 
 impl Manifest {
-    pub fn new(db: &dyn SemanticGroup, crate_ids: &[CrateId]) -> Self {
-        let mut manifest = Manifest { components: vec![], systems: vec![] };
+    pub fn new(
+        db: &dyn SemanticGroup,
+        crate_ids: &[CrateId],
+        compiled_classes: Vec<(SmolStr, FieldElement)>,
+    ) -> Self {
+        let mut manifest = Manifest::default();
+
+        let world = get_compiled_class_hash(&compiled_classes, "World").unwrap_or_else(|| {
+            panic!("World contract not found. Did you include `dojo` as a dependency?");
```

Comment:
```suggestion
            panic!("World contract not found. Did you include `dojo_core` as a dependency?");
```

This crate got renamed